arbnode/timeboost_delayed_sequencer.go

```diff
@@ -44,86 +41,65 @@ var TestTimeboostDelayedSequencerConfig = TimeboostDelayedSequencerConfig{
 	Enable: false,
 }
 
-func NewTimeboostDelayedSequencer(reader *InboxReader, exec execution.ExecutionSequencer, config TimeboostDelayedSequencerConfigFetcher) (*TimeboostDelayedSequencer, chan gethexec.DelayedMessageCommand, error) {
-	delayedChannel := make(chan gethexec.DelayedMessageCommand, 1)
+func NewTimeboostDelayedSequencer(reader *InboxReader, exec execution.ExecutionSequencer, config TimeboostDelayedSequencerConfigFetcher) (*TimeboostDelayedSequencer, error) {
 	d := &TimeboostDelayedSequencer{
-		inbox:              reader.Tracker(),
-		reader:             reader,
-		exec:               exec,
-		config:             config,
-		delayedMessageChan: delayedChannel,
+		inbox:  reader.Tracker(),
+		reader: reader,
+		exec:   exec,
+		config: config,
 	}
-	return d, delayedChannel, nil
+	return d, nil
 }
 
 func (d *TimeboostDelayedSequencer) getDelayedMessagesRead() (uint64, error) {
 	return d.exec.NextDelayedMessageNumber()
 }
 
-func (d *TimeboostDelayedSequencer) sequence(ctx context.Context, delayedCount uint64) error {
+func (d *TimeboostDelayedSequencer) SequenceDelayedMessages(ctx context.Context, delayedCount uint64) ([]*decentralizedtimeboost.TimeboostDelayedMessage, error) {
 	config := d.config()
 	if !config.Enable {
-		return nil
+		return nil, nil
 	}
 
 	startPos, err := d.getDelayedMessagesRead()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// Retrieve all finalized delayed messages
 	pos := startPos
-	var messages []*arbostypes.L1IncomingMessage
+	var messages []*decentralizedtimeboost.TimeboostDelayedMessage
 	for pos < delayedCount {
 		msg, _, _, err := d.inbox.GetDelayedMessageAccumulatorAndParentChainBlockNumber(ctx, pos)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		err = msg.FillInBatchGasCost(func(batchNum uint64) ([]byte, error) {
 			data, _, err := d.reader.GetSequencerMessageBytes(ctx, batchNum)
 			return data, err
 		})
 		if err != nil {
-			return err
+			return nil, err
 		}
-		messages = append(messages, msg)
+		delayedMsg := &decentralizedtimeboost.TimeboostDelayedMessage{
+			Message: msg,
+			Pos:     pos + 1,
```

Comment:
would this not be an issue when pos == delayedCount -1? Then the delayed message here will become delayedCount?